packages/jest-cli/src/runTest.js

```diff
@@ -27,24 +27,25 @@ function runTest(path: Path, config: Config, resolver: Resolver) {
   /* $FlowFixMe */
   const ModuleLoader = require(config.moduleLoader || 'jest-runtime');
 
-  const env = new TestEnvironment(config);
   const TestConsole =
     config.verbose
       ? Console
       : (config.silent
         ? NullConsole
         : BufferedConsole
       );
-  const testConsole = env.global.console = new TestConsole(
-    config.useStderr ? process.stderr : process.stdout,
-    process.stderr,
-    (type, message) => getConsoleOutput(
-      config.rootDir,
-      !!config.verbose,
-      // 4 = the console call is burried 4 stack frames deep
-      BufferedConsole.write([], type, message, 4),
-    ),
-  );
+  const createCustomConsole = (stackFrameDepth: number): Object => {
```

Comment:
I think `createConsole` should be good enough of a name, it implies that it is "custom".